src/Microsoft.AspNet.Mvc/RazorPreCompileModule.cs

```diff
@@ -84,6 +76,27 @@ public void AfterCompile(IAfterCompileContext context)
         {
         }
 
+        // REVIEW: Welcome back to life delegating SP!!!
+        private class WrappingServiceProvider : IServiceProvider
+        {
+            private readonly IServiceProvider _inner;
+            private readonly IServiceProvider _outer;
```

Comment:
I have no idea what these names mean without reading the code.  use something like `_override` and `_fallback`